niworkflows/interfaces/cifti.py

```diff
@@ -68,112 +78,141 @@ class GenerateCifti(SimpleInterface):
+            if all(
+                target in targets for target in
+                [self.inputs.surface_target, self.inputs.volume_target]
+            ):
                 space = sp
         if space is None:
             raise NotImplementedError
 
         variant_key = os.path.abspath('dtseries_variant.json')
+        out_json = {
+            'space': space,
+            'surface': variants[space][0],
+            'volume': variants[space][1]
+        }
+        if self.inputs.surface_target == 'fsLR':
+            # 91k == 2mm resolution, 170k == 1.6mm resolution
+            out_json['grayordinates'] = '91k' if self.inputs.density == '32k' else '170k'
         with open(variant_key, 'w') as fp:
-            json.dump({space: variants[space]}, fp)
-        return variant_key, space
+            json.dump(out_json, fp)
+        self._results['variant_key'] = variant_key
+        self._results['variant'] = space
+        return variants[space]
 
     def _fetch_data(self):
-        """Convert inputspec to files."""
-        if (self.inputs.surface_target == "fsnative" or
-                self.inputs.volume_target != "MNI152NLin2009cAsym"):
-            # subject space is not support yet
-            raise NotImplementedError
-
-        annotation_files = sorted(glob(os.path.join(self.inputs.subjects_dir,
-                                                    self.inputs.surface_target,
-                                                    'label',
-                                                    '*h.aparc.annot')))
-        if not annotation_files:
-            raise IOError("Freesurfer annotations for %s not found in %s" % (
-                          self.inputs.surface_target, self.inputs.subjects_dir))
-
-        label_file = str(get_template(
-            'MNI152NLin2009cAsym', resolution=2, desc='DKT31', suffix='dseg'))
+        """Converts inputspec to files"""
+        if (
+            self.inputs.surface_target not in self._surfaces or
+            self.inputs.volume_target not in self._volumes
+        ):
+            raise NotImplementedError(
+                "Target space (surface: {0}, volume: {1}) is not supported".format(
+                    self.inputs.surface_target, self.inputs.volume_target
+                )
+            )
+
+        tpl_kwargs = {'suffix': 'dseg'}
+        annotation_files = None
+        if self.inputs.volume_target == "MNI152NLin2009cAsym":
+            tpl_kwargs.update({
+                'resolution': '2',
+                'desc': 'DKT31',
+            })
+            annotation_files = sorted(
+                glob(os.path.join(self.inputs.subjects_dir,
+                                  self.inputs.surface_target,
+                                  'label',
+                                  '*h.aparc.annot'))
+            )
+
+        elif self.inputs.volume_target == 'MNI152NLin6Asym':
+            res = '2' if self.inputs.density == '32k' else '6'
```

Comment:
Perhaps a stricter value checking?
```suggestion
            res = {'59k': '6', '32k': '2}[self.inputs.density]
```